extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -7,251 +7,64 @@ import * as mssql from 'mssql';
-			, this.deploymentOptions.ignoreComments.displayName
-			, this.deploymentOptions.ignoreColumnCollation.displayName
-			, this.deploymentOptions.ignoreAuthorizer.displayName
-			, this.deploymentOptions.ignoreAnsiNulls.displayName
-			, this.deploymentOptions.generateSmartDefaults.displayName
-			, this.deploymentOptions.dropStatisticsNotInSource.displayName
-			, this.deploymentOptions.dropRoleMembersNotInSource.displayName
-			, this.deploymentOptions.dropPermissionsNotInSource.displayName
-			, this.deploymentOptions.dropObjectsNotInSource.displayName
-			, this.deploymentOptions.ignoreColumnOrder.displayName
-			, this.deploymentOptions.ignoreTablePartitionOptions.displayName
-			, this.deploymentOptions.doNotEvaluateSqlCmdVariables.displayName
-			, this.deploymentOptions.disableParallelismForEnablingIndexes.displayName
-			, this.deploymentOptions.disableIndexesForDataPhase.displayName
-			, this.deploymentOptions.restoreSequenceCurrentValue.displayName
-			, this.deploymentOptions.rebuildIndexesOfflineForDataPhase.displayName
-			, this.deploymentOptions.preserveIdentityLastValues.displayName
-			, this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled.displayName
-			, this.deploymentOptions.allowExternalLibraryPaths.displayName
-			, this.deploymentOptions.allowExternalLanguagePaths.displayName
-			, this.deploymentOptions.hashObjectNamesInLogs.displayName
-			, this.deploymentOptions.doNotDropWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreDatabaseWorkloadGroups.displayName
-			, this.deploymentOptions.doNotDropDatabaseWorkloadGroups.displayName].sort();
+	constructor(private defaultOptions: mssql.DeploymentOptions) {
+		this.deploymentOptions = { ...this.defaultOptions };
 	}
 
 	/**
-	 * Gets the options checkbox check value
-	 * @returns string[][]
+	 * Initialize options data from deployment options for table component
+	 * Also preparing optionsValueLookup Map holding onchange checkbox values and optionsNameAndPropMap to hold property name for the option
+	 * Returns data as [booleanValue, optionName]
 	 */
-	public getOptionsData(): string[][] {
-		let data: any = [];
-		this.optionsLookup = {};
-		this.optionsLabels.forEach(l => {
-			let checked: boolean = this.getDeployOptionUtil(l);
-			data.push([checked, l]);
-			this.optionsLookup[l] = checked;
+	public initializeOptionsData(): any[][] {
+		let data: any[][] = [];
+		Object.entries(this.deploymentOptions.booleanOptionsDictionary).forEach(option => {
+			const optionDisplayName = option[1].displayName;
+			const propertyName = option[0];
+			const checkedValue = option[1].value;
+			// push to table array
+			data.push([checkedValue, optionDisplayName]);
+			// push to optionsNameAndPropMap
+			this.optionsNameAndPropMap[optionDisplayName] = propertyName;
+			// push to optionsValueLookup
+			this.optionsValueLookup[optionDisplayName] = checkedValue;
```

Comment:
does there need to be 2 separate lookup objects that use `optionDisplayName` as the key? Why not have one that has both the `propertyName` and `checkedValue`?

Reply:
Now I get the comment of @chlafreniere, actually no need of both, I can merge them!